### PR TITLE
omit unnecessary spaces around <b>, <em>, and <strike>

### LINKED ISSUE #324: Unnecessary spaces around <b></b>
The html code `X<b>one</b>, two` is translated into `X **one** , two\n\n`, but should be translated into `X**one**, two\n\n` instead (without spaces around **one**).

- Version by `html2text --version`: 2020.1.16

- Test script

```
import html2text
h = html2text.HTML2Text()
h.handle('X<b>one</b>, two')
```
- Python version `python --version`: 3.8.2

### PR DESCRIPTION
**Pull request by snarfed**
fixes #324

### CODE REVIEW

File: html2text/__init__.py [hunk 1]
```diff
@@ -420,7 +421,11 @@ def no_preceding_space(self: HTML2Text) -> bool:
                 self.stressed = True
 
         if tag in ["strong", "b"] and not self.ignore_emphasis:
-            if start and no_preceding_space(self):
+            if (
```
Comment (Alir3z4) [inline html2text/__init__.py:424]: Would be great to have some reasoning documented here (_documenting WHY_) so reading code would be easier.

File: html2text/__init__.py [hunk 2]
```diff
@@ -404,13 +405,13 @@ def handle_tag(
...
                 self.p()
 
-        def no_preceding_space(self: HTML2Text) -> bool:
-            return bool(
-                self.preceding_data and re.match(r"[^\s]", self.preceding_data[-1])
-            )
-
         if tag in ["em", "i", "u"] and not self.ignore_emphasis:
-            if start and no_preceding_space(self):
+            if (
```
Comment (Alir3z4) [inline html2text/__init__.py:409]: Same as https://github.com/Alir3z4/html2text/pull/342/files#r591848916

File: html2text/__init__.py (hunk 1 above)
Comment (snarfed) [reply html2text/__init__.py:433]: Good call! Thanks, done.

File: html2text/__init__.py (hunk 2 above)
Comment (snarfed) [reply html2text/__init__.py:414]: Sure! Done.

**Comment by djdembeck** [comment]
Noticed these problems as well. Any chance this can be merged?

**Comment by Alir3z4** [comment]
@snarfed @djdembeck just merged this in and I noticed it failed on master.
Could you please let me know if you could troubleshoot?

**Comment by Alir3z4** [comment]
The failure can be seen on travis build report https://travis-ci.org/github/Alir3z4/html2text/builds/769590651?utm_source=github_status&utm_medium=notification

**Comment by snarfed** [comment]
Sure! Done, in #357. Looks like this one collided with #333, which was merged after this PR passed CI but before it was merged.

**Comment by Alir3z4** [comment]
Excellent!

**Pull request merged**